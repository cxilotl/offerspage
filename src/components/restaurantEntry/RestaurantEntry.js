import React from 'react';
import PropTypes from 'prop-types';
import cssStyles from './RestaurantEntry.module.scss';

const RestaurantEntry = ({ name, description, priceRange }) => {
  return(
    <div className={ cssStyles.layout }>
      <h3>{ name }</h3>
      <p><span>{ description }</span> . <span>{ priceRange }</span></p>
    </div>
  );
};

RestaurantEntry.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  priceRange: PropTypes.string
};

export default RestaurantEntry;