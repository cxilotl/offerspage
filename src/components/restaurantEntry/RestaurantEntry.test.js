import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RestaurantEntry from './RestaurantEntry';

describe('RestaurantEntry', () => {
  const restaurantInfo = {
    name: 'Benito\'s Hat',
    description: `Mexican . Burritos`,
    priceRange: '£'
  };

  it('Snapshot', () => {
    const component = renderer.create(
      <RestaurantEntry
        name={ restaurantInfo.name }
        description={ restaurantInfo.description }
        priceRange={ restaurantInfo.priceRange }
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Given some Restaurant information, it', () => {

    it('Should render the name, the description, and the price range of the restaurant', () => {
      const { getByText } = render(
        <RestaurantEntry
          name={ restaurantInfo.name }
          description={ restaurantInfo.description }
          priceRange={ restaurantInfo.priceRange }
        />
      );

      const name = getByText(/Benito's Hat/i);
      expect(name).toBeInTheDocument();

      const description = getByText(/Mexican/i);
      expect(description.textContent).toBe(`Mexican . Burritos`);

      const priceRange = getByText(/£/i);
      expect(priceRange).toBeInTheDocument();
    });
  });
});