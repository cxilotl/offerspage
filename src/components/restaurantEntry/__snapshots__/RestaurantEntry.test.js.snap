// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantEntry Snapshot 1`] = `
<div
  className="layout"
>
  <h3>
    Benito's Hat
  </h3>
  <p>
    <span>
      Mexican . Burritos
    </span>
     . 
    <span>
      Â£
    </span>
  </p>
</div>
`;
